lib/buffer_update_component.coffee

```diff
@@ -5,11 +6,14 @@ module.exports =
 class BufferUpdateComponent
   constructor: (@editor, @analysisAPI) ->
     @events = new CompositeDisposable()
-    @events.add @editor.onDidStopChanging =>
+    updateFile = =>
       descriptors = @editor.getRootScopeDescriptor()
       # We only support pure Dart files for now
       return unless Utils.isCompatible(editor)
       @analysisAPI.updateFile @editor.getPath(), @editor.getText()
 
+    @events.add @editor.onDidStopChanging _.debounce(updateFile, 250,
+      maxWait: 1000)
```

Comment:
That's a pretty long time to wait, especially if the user isn't really thinking about the code in question and just typing along. What if someone is doing some rote stuff like a map reduce on a List? They'll be way into it before they notice they've made a typo.